ocp_utilities/monitoring.py

```diff
@@ -0,0 +1,199 @@
+    """
+
+    def __init__(
+        self, namespace=None, resource_name=None, client=None, verify_ssl=None
+    ):
+        self.namespace = namespace or "openshift-monitoring"
+        self.resource_name = resource_name or "prometheus-k8s"
+        self.client = client or get_client()
+        self.api_v1 = "/api/v1"
+        self.verify_ssl = True
+
+        route = Route(
+            namespace=self.namespace, name=self.resource_name, client=self.client
+        ).instance.spec.host
+
+        self.api_url = f"https://{route}"
+        self.headers = self._get_headers()
+        self.scrapeInterval = self.getscrapeInterval()
+
+    def _get_headers(self):
+        """Uses the Prometheus serviceaccount to get an access token for OAuth"""
+
+        LOGGER.info("Setting Prometheus headers and Obtaining OAuth token")
+
+        prometheus_sa = ServiceAccount(
+            namespace=self.namespace, name=self.resource_name, client=self.client
+        )
+
+        secret_name = prometheus_sa.instance.imagePullSecrets[0].name
+        secret = Secret(namespace=self.namespace, name=secret_name, client=self.client)
+
+        token = secret.instance.metadata.annotations["openshift.io/token-secret.value"]
+
+        return {"Authorization": f"Bearer {token}"}
+
+    def _get_response(self, query):
+        response = requests.get(
+            f"{self.api_url}{query}", headers=self.headers, verify=self.verify_ssl
+        )
+
+        try:
+            return json.loads(response.content)
+        except JSONDecodeError as json_exception:
+            LOGGER.error(
+                "Exception converting query response to JSON: "
+                f"exc={json_exception} response_status_code={response.status_code} response={response.content}"
+            )
+            raise
+
+    def query(self, query):
+        """get the prometheus query result
+
+        Args:
+            query(str): promthetheus query string
+
+        Returns:
+            dict: query result
+        """
+        return self._get_response(query=f"{self.api_v1}/query?query={query}")
+
+    def get_alert(self, alert_name):
+        """
+        Get alert by alert anem if it's an active alert
+
+        Args:
+             alert(str): alert name
+
+        Examples:
+             result = prometheus.get_alert(alert='WatchDog')
+
+        Returns:
+             dict: dictiionary containing alert name as key and alert data as value
+        """
+        response = self.alerts
+        LOGGER.info(f" response activer alerts : {response}")
+        matched_alerts = {}
+        alert_list = []
+        for alert in response["data"]["alerts"]:
+            if alert and alert["labels"]["alertname"] == alert_name:
+                alert_list.append(alert)
+        if not alert_list:
+            LOGGER.error(f"No Matching alerts found for alert: {alert_name}")
+        else:
+            matched_alerts[alert_name] = alert_list
+            return matched_alerts
```

Comment:
This will return `None` if no match is found, then may be you should consider returning the list instead of a dict with one key if a match is found.

Reply:
I tested it it in case of no match found it's not returning None, previously it was but not after this changed
I'll check again

Reply:
this is the output from after the change

`2023-04-04T19:41:57.200206 ocp_utilities.monitoring INFO  response activer alerts : {'status': 'success', 'data': {'alerts': [{'labels': {'alertname': 'PodSecurityViolation', 'namespace': 'openshift-kube-apiserver', 'policy_level': 'restricted', 'severity': 'info'}, 'annotations': {'description': 'A workload (pod, deployment, deamonset, ...) was created somewhere in the cluster but it did not match the PodSecurity "restricted" profile defined by its namespace either via the cluster-wide configuration (which triggers on a "restricted" profile violations) or by the namespace local Pod Security labels. Refer to Kubernetes documentation on Pod Security Admission to learn more about these violations.', 'summary': "One or more workloads users created in the cluster don't match their Pod Security profile"}, 'state': 'firing', 'activeAt': '2023-04-04T16:17:18.474370795Z', 'value': '2.0026335566940867e+00'}, {'labels': {'alertname': 'ClusterNotUpgradeable', 'condition': 'Upgradeable', 'endpoint': 'metrics', 'name': 'version', 'namespace': 'openshift-cluster-version', 'severity': 'info'}, 'annotations': {'description': "In most cases, you will still be able to apply patch releases. Reason AdminAckRequired. For more information refer to 'oc adm upgrade' or https://console-openshift-console.apps.vs-test-80.wih7.s1.devshift.org/settings/cluster/.", 'summary': 'One or more cluster operators have been blocking minor version cluster upgrades for at least an hour.'}, 'state': 'firing', 'activeAt': '2023-04-04T15:47:10.071245183Z', 'value': '0e+00'}, {'labels': {'alertname': 'Watchdog', 'namespace': 'openshift-monitoring', 'severity': 'none'}, 'annotations': {'description': 'This is an alert meant to ensure that the entire alerting pipeline is functional.\nThis alert is always firing, therefore it should always be firing in Alertmanager\nand always fire against a receiver. There are integrations with various notification\nmechanisms that send a notification when this alert is not firing. For example the\n"DeadMansSnitch" integration in PagerDuty.\n', 'summary': 'An alert that should always be firing to certify that Alertmanager is working properly.'}, 'state': 'firing', 'activeAt': '2023-04-04T15:47:58.572280542Z', 'value': '1e+00'}]}}
2023-04-04T19:41:57.200793 ocp_utilities.monitoring ERROR No Matching alerts found for alert: KubePodNotScheduled

TEST: TestClustrAlerts.test_no_alerts_firing_on_healthy_cluster STATUS: PASSED
`

Reply:
Python does implicit return in case you are not returning anything (your are not returning anything in case no match is found): https://realpython.com/python-return-statement/#implicit-return-statements

I don't understand the need to create a dictionary with one key - alert name specified by the caller. Why not return the list alert_list directly?

Reply:
can update it to raise error in case no alert found